### PR TITLE
cleanup: Fetch wraps

### PR DESCRIPTION
**Pull request by lediur**
Merge after #65

### CODE REVIEW

File: src/api/fetch.tsx [hunk 1]
```diff
@@ -45,19 +22,35 @@ export function getAnalyticsParams(): {
 
 export function buildExtensionHeaders(
   user: string | undefined,
-  installationId: string
+  installationId: string | undefined
 ) {
   return {
     "X-Secarta-GitHub-User": user || `anonymous-${installationId}`,
-    "X-R2C-Extension-Installation-Id": installationId,
+    "X-R2C-Extension-Installation-Id": installationId || "not-generated",
```
Comment (dlukeomalley) [inline src/api/fetch.tsx]: Is this a bug fix?

File: src/content/index.tsx [hunk 2]
```diff
@@ -78,150 +80,167 @@ const ShareIcon: React.SFC = () => (
...
-    if (isRepositoryPrivate() || installationId === "not-generated") {
+    if (isRepositoryPrivate()) {
       return null;
     }
 
     if (!isGitHubSlug(this.repoSlug)) {
       return null;
     }
 
+    if (installationId == null) {
```
Comment (dlukeomalley) [inline src/content/index.tsx]: What does this condition protect against? Shouldn't a user always have an installationId?

File: src/content/index.tsx [hunk 3]
```diff
@@ -78,150 +80,167 @@ const ShareIcon: React.SFC = () => (
...
+                    panel={
+                      <Discussion user={user} installationId={installationId} />
+                    }
+                  />
+                  <ActionButton
+                    id="flag"
+                    title="Flag an issue with this project"
+                    icon={<ReportIcon />}
+                    selected={
+                      voteData != null ? voteData.currentVote === "down" : false
```
Comment (dlukeomalley) [inline src/content/index.tsx]: Perhaps a follow-up PR, this shouldn't be `"down"` anymore.

File: src/content/github/dom.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    const userLoginMetaTags = document.getElementsByName("user-login");
+
+    if (userLoginMetaTags == null || userLoginMetaTags.length === 0) {
+      return undefined;
+    }
+
+    const user = userLoginMetaTags[0].getAttribute("content");
+
+    if (user != null && user !== "") {
+      setGitHubUser(user);
```
Comment (dlukeomalley) [inline src/content/github/dom.ts]: Does this save the user locally? I've noticed that feed clicks never have users associated with them, only installation ids.

File: src/content/PackageCopyBox.tsx [hunk 5]
```diff
@@ -12,6 +12,7 @@ import {
 import { IconNames } from "@blueprintjs/icons";
 import { ItemRenderer, Select } from "@blueprintjs/select";
 import { l } from "@r2c/extension/analytics";
+import { ApiFetch } from "@r2c/extension/api/fetch";
```
Comment (DrewDennison) [inline src/content/PackageCopyBox.tsx]: Shouldn't we put an index.ts in api so it's just import from api?

Comment (lediur) [reply src/content/PackageCopyBox.tsx]: good idea - `fetch` was the old file but can move it out

File: src/content/index.tsx (hunk 3 above)
Comment (lediur) [reply src/content/index.tsx]: yeah let's follow up

File: src/content/index.tsx (hunk 2 above)
Comment (lediur) [reply src/content/index.tsx]: `installationId` is asynchronously fetched / generated, so there is a possibility that the value is undefined. Previously this was represented as `not-generated`. I've changed that to make it explicit that `installationId` hasn't been created yet.

File: src/api/fetch.tsx (hunk 1 above)
Comment (lediur) [reply src/api/fetch.tsx]: from another comment

> Previously this was represented as not-generated. I've changed that to make it explicit that `installationId` hasn't been created yet.

**Pull request merged**